### PR TITLE
chore(bench): measure key sizes and zk proof sizes

### PR DESCRIPTION
**Pull request by soonum**

### CODE REVIEW

File: tfhe/benches/integer/zk_pke.rs [hunk 1]
```diff
@@ -178,6 +181,23 @@ fn pke_zk_verify(c: &mut Criterion, results_file: &Path) {
                     .build_with_proof_packed(public_params, compute_load)
                     .unwrap();
 
+                let proof_serialized = bincode::serialize(&ct1).unwrap();
+
+                println!("proof size: {}", proof_serialized.len());
+
+                let test_name = format!("zk::proof_sizes::{param_name}_{bits}_bits_packed");
```
Comment (IceTDrinker) [inline tfhe/benches/integer/zk_pke.rs:188]: ah damn it forgot the type of proof here 🙄 I'll update the last commit

File: tfhe/src/shortint/key_switching_key/mod.rs [hunk 2]
```diff
@@ -733,4 +733,108 @@ impl CompressedKeySwitchingKey {
...
+    /// Construct a [`CompressedKeySwitchingKey`] from its constituents.
+    ///
+    /// # Panics
+    ///
+    /// Panics if the provided raw parts are not compatible with each other, i.e.:
+    ///
+    /// if the provided source [`CompressedServerKey`] ciphertext
+    /// [`LweDimension`](`crate::core_crypto::commons::parameters::LweDimension`) does not match the
+    /// input [`LweDimension`](`crate::core_crypto::commons::parameters::LweDimension`) of the
+    /// [`LweKeyswitchKeyOwned`] in the provided [`CompressedKeySwitchingKeyMaterial`] or if the
```
Comment (IceTDrinker) [inline tfhe/src/shortint/key_switching_key/mod.rs:763]: LweKeyswitchKeyOwned -> SeededLweKeyswitchKeyOwned 

I believe ?

File: tfhe/src/shortint/key_switching_key/mod.rs [hunk 3]
```diff
@@ -733,4 +733,108 @@ impl CompressedKeySwitchingKey {
...
+    /// provided destination [`ServerKey`] ciphertext
```
Comment (IceTDrinker) [inline tfhe/src/shortint/key_switching_key/mod.rs:764]: CompressedServerKey I believe ?

File: tfhe/src/shortint/key_switching_key/mod.rs [hunk 4]
```diff
@@ -733,4 +733,108 @@ impl CompressedKeySwitchingKey {
...
+                    src_lwe_dimension,
+                    key_switching_key_material
+                        .key_switching_key
+                        .input_key_lwe_dimension(),
+                    "Mismatch between the source ServerKey ciphertext LweDimension ({:?}) \
+                    and the LweKeyswitchKey input LweDimension ({:?})",
+                    src_lwe_dimension,
+                    key_switching_key_material
+                        .key_switching_key
+                        .input_key_lwe_dimension(),
```
Comment (IceTDrinker) [inline tfhe/src/shortint/key_switching_key/mod.rs:787]: CompressedServerKey and SeededLweKeyswitchKey

I think

File: tfhe/src/shortint/key_switching_key/mod.rs [hunk 5]
```diff
@@ -733,4 +733,108 @@ impl CompressedKeySwitchingKey {
...
+                assert_eq!(
+                    src_server_key.ciphertext_modulus, dest_server_key.ciphertext_modulus,
+                    "Mismatch between the source ServerKey CiphertextModulus ({:?}) \
+                    and the destination ServerKey CiphertextModulus ({:?})",
+                    src_server_key.ciphertext_modulus, dest_server_key.ciphertext_modulus,
+                );
+            }
+            None => assert!(
+                key_switching_key_material.cast_rshift >= 0,
+                "Trying to build a shortint::KeySwitchingKey with a negative cast_rshift \
```
Comment (IceTDrinker) [inline tfhe/src/shortint/key_switching_key/mod.rs:799]: CompressedKeyswitchingKey

File: tfhe/src/shortint/key_switching_key/mod.rs [hunk 6]
```diff
@@ -733,4 +733,108 @@ impl CompressedKeySwitchingKey {
...
+        };
+
+        assert_eq!(
+            dst_lwe_dimension,
+            key_switching_key_material
+                .key_switching_key
+                .output_key_lwe_dimension(),
+            "Mismatch between the destination ServerKey ciphertext LweDimension ({:?}) \
+            and the LweKeyswitchKey output LweDimension ({:?})",
+            dst_lwe_dimension,
```
Comment (IceTDrinker) [inline tfhe/src/shortint/key_switching_key/mod.rs:816]: compressed names

File: tfhe/src/shortint/key_switching_key/mod.rs [hunk 7]
```diff
@@ -733,4 +733,108 @@ impl CompressedKeySwitchingKey {
...
+        );
+        assert_eq!(
+            key_switching_key_material
+                .key_switching_key
+                .ciphertext_modulus(),
+            dest_server_key.ciphertext_modulus,
+            "Mismatch between the LweKeyswitchKey CiphertextModulus ({:?}) \
+            and the destination ServerKey CiphertextModulus ({:?})",
+            key_switching_key_material
+                .key_switching_key
```
Comment (IceTDrinker) [inline tfhe/src/shortint/key_switching_key/mod.rs:829]: compressed names

**Comment by IceTDrinker** [comment]
I will take care of these edits

File: tfhe/examples/utilities/shortint_key_sizes.rs [hunk 8]
```diff
@@ -128,6 +135,150 @@ fn client_server_key_sizes(results_file: &Path) {
     }
 }
 
+fn measure_serialize_size<T: serde::Serialize, P: Into<CryptoParametersRecord<u64>> + Clone>(
```
Comment (nsarlin-zama) [inline tfhe/examples/utilities/shortint_key_sizes.rs:138]: maybe `measure_serialized_size` ?

**Review by nsarlin-zama — COMMENTED** [review]
One bench computes the size of the in-memory rust object while the other computes the size of the bincode serialized data. This could be uniformized later or at least specified in the function names.

File: tfhe/examples/utilities/shortint_key_sizes.rs (hunk 8 above)
Comment (IceTDrinker) [reply tfhe/examples/utilities/shortint_key_sizes.rs:138]: indeed

Comment (IceTDrinker) [reply tfhe/examples/utilities/shortint_key_sizes.rs]: done

**Pull request merged**